### DISCUSSION TITLE
是否可以统一把 vant-cli2.x  和   vant-cli3.x  升级到 vant-cli4.x 方案

### DISCUSSION THREAD

**Discussion by hxzAugustinus** [youzan/vant#10189] [Ideas]
目前我的理解：使用vant-cli4 打包vue2 和 vue3  差异部分就2个模块  site模板   和  vite plugin 
使用vue-demi 重构  vant-cli/site部分 ，可以抹平 site 里的依赖 和外面的依赖一致
 vite 的 plugin 里的差异可以通过初始化 询问生成配置项处理 ，比较简单

这样做好处 ：
1 . 可以统一2个分支版本 ，只用维护一个分支；
2.  vue2 也可以 无痛 升级到  vue-cli4的 vite ；

不知道是否可行？

**Comment by chenjiahan** [comment]
理论上可行，但我们没有计划在 4.x 版本里去兼容 Vue 2 组件库，Vue 2 组件库继续用原来的版本也没什么问题。
